libcloud/dns/drivers/auroradns.py

```diff
@@ -0,0 +1,260 @@
+            params = {}
+
+        if method in ("POST", "PUT"):
+            headers = {'Content-Type': 'application/json; charset=UTF-8'}
+
+        t = datetime.datetime.utcnow()
+        timestamp = t.strftime('%Y%m%dT%H%M%SZ')
+
+        headers['X-AuroraDNS-Date'] = timestamp
+        headers['Authorization'] = self.gen_auth_header(self.user_id, self.key,
+                                                        method, action,
+                                                        timestamp)
+
+        return super(AuroraDNSConnection, self).request(action=action,
+                                                        params=params,
+                                                        data=data,
+                                                        method=method,
+                                                        headers=headers)
+
+
+class AuroraDNSDriver(DNSDriver):
+    name = 'AuroraDNS'
+    website = 'https://www.pcextreme.nl/en/aurora/dns'
+    connectionCls = AuroraDNSConnection
+
+    RECORD_TYPE_MAP = {
+        RecordType.A: 'A',
+        RecordType.AAAA: 'AAAA',
+        RecordType.CNAME: 'CNAME',
+        RecordType.MX: 'MX',
+        RecordType.NS: 'NS',
+        RecordType.SOA: 'SOA',
+        RecordType.SPF: 'SPF',
+        RecordType.SRV: 'SRV',
+        RecordType.TXT: 'TXT',
+    }
+
+    def __res_to_record(self, zone, record):
+        if len(record['name']) == 0:
+            name = None
+        else:
+            name = record['name']
+
+        extra = {}
+        extra['created'] = record['created']
+        extra['modified'] = record['modified']
+        extra['disabled'] = record['disabled']
+        extra['ttl'] = record['ttl']
+        extra['prio'] = record['prio']
+
+        return Record(id=record['id'], name=name, type=record['type'],
+                      data=record['content'], zone=zone, driver=self,
+                      extra=extra)
+
+    def __res_to_zone(self, zone):
+        return Zone(id=zone['id'], domain=zone['name'], type=DEFAULT_ZONE_TYPE,
+                    ttl=DEFAULT_ZONE_TTL, driver=self,
+                    extra={'created': zone['created'],
+                           'servers': zone['servers'],
+                           'account_id': zone['account_id'],
+                           'cluster_id': zone['cluster_id']})
+
+    def __merge_extra_data(self, rdata, extra):
+        if extra is not None:
+            for param in VALID_RECORD_PARAMS_EXTRA:
+                if param in extra:
+                    rdata[param] = extra[param]
+
+        return rdata
+
+    def list_zones(self):
+        zones = []
+
+        res = self.connection.request('/zones')
+        for zone in res.parse_body():
+            zones.append(self.__res_to_zone(zone))
+
+        return zones
+
+    def list_records(self, zone):
+        records = []
+        res = self.connection.request('/zones/{0}/records'.format(zone.id))
+
+        for record in res.parse_body():
```

Comment:
You shouldn't call `res.parse_body()` yourself unless it's necessary.

This function is already called in the constructor and parsed body is available via `res.object` variable.